newsroom/topics/views.py

```diff
@@ -1,83 +1,108 @@
 from bson import ObjectId
+from typing import Optional
+from pydantic import BaseModel, ValidationError
 
 from superdesk.core import json, get_app_config
-from superdesk import get_resource_service
-from superdesk.flask import jsonify, abort, url_for, session
+from superdesk.flask import abort, url_for, session
+from superdesk.core.web import Request, Response
 
 from newsroom.types import Topic
-from newsroom.topics import blueprint
-from newsroom.topics.topics import get_user_topics as _get_user_topics, auto_enable_user_emails
-from newsroom.auth import get_user, get_user_id
-from newsroom.decorator import login_required
-from newsroom.utils import get_json_or_400, get_entity_or_404
 from newsroom.email import send_user_email
+from newsroom.decorator import login_required
+from newsroom.auth import get_user, get_user_id
+from newsroom.topics.topics_async import get_user_topics as _get_user_topics, auto_enable_user_emails
+from newsroom.utils import get_json_or_400, get_entity_or_404, response_from_validation
 from newsroom.notifications import (
     push_user_notification,
     push_company_notification,
     save_user_notifications,
     UserNotification,
 )
+from .topics_async import topic_endpoints, TopicService, TopicResourceModel
+from newsroom.users.service import UsersService
+
+
+class RouteArguments(BaseModel):
+    user_id: Optional[str] = None
+    topic_id: Optional[str] = None
 
 
-@blueprint.route("/users/<_id>/topics", methods=["GET"])
+@topic_endpoints.endpoint("/users/<string:user_id>/topics", methods=["GET"])
 @login_required
-async def get_topics(_id):
+async def get_topics(args: RouteArguments, params: None, request: Request):
     """Returns list of followed topics of given user"""
-    if session["user"] != str(_id):
+    if session["user"] != str(args.user_id):
         abort(403)
-    return jsonify({"_items": _get_user_topics(_id)}), 200
+    topics = await _get_user_topics(args.user_id)
+    return Response({"_items": topics}, 200, ())
```

Comment:
```suggestion
    return success_response({"_items": topics})
```

most of these 200 status code responses have empty headers. I created this small helper function in case you want to reuse it :) 

https://github.com/superdesk/newsroom-core/blob/6d96d26cec92684d7879f91fb6c1ebe83eb409ef/newsroom/utils.py#L150-L160

Reply:
Also here, once https://github.com/superdesk/superdesk-core/pull/2693 is merged, you can disregard this suggestion and simply use `return Response({"_items": topics})` as `status_code` and `headers` will have default value.